code/modules/resleeving/sleevecard.dm

```diff
@@ -1,60 +1,33 @@
-/obj/item/device/radio/sleevecard
-	canhear_range = 0
-
-/obj/item/device/radio/sleevecard/tgui_state(mob/user)
-	return GLOB.tgui_always_state
-
-/obj/item/device/sleevecard
+/obj/item/device/paicard/sleevecard
 	name = "sleevecard"
 	desc = "This KHI-upgraded pAI module has enough capacity to run a whole mind of human-level intelligence."
 	catalogue_data = list(/datum/category_item/catalogue/information/organization/khi,
 						/datum/category_item/catalogue/technology/resleeving)
-
-	icon = 'icons/obj/pda.dmi'
-	icon_state = "pai"
-	item_state = "electronic"
-
-	w_class = ITEMSIZE_SMALL
-	slot_flags = SLOT_BELT
 	origin_tech = list(TECH_DATA = 2)
 	show_messages = 0
 
-	var/obj/item/device/radio/sleevecard/radio
-	var/mob/living/silicon/infomorph/infomorph
-	var/current_emotion = 1
-
 	matter = list(MAT_STEEL = 4000, MAT_GLASS = 4000)
 
-/obj/item/device/sleevecard/relaymove(var/mob/user, var/direction)
-	if(user.stat || user.stunned)
-		return
-	var/obj/item/weapon/rig/rig = src.get_rig()
-	if(istype(rig))
-		rig.forced_move(direction, user)
-
-/obj/item/device/sleevecard/New()
-	..()
-	add_overlay("pai-off")
-	radio = new(src)
-
-/obj/item/device/sleevecard/Destroy()
-	if(!isnull(infomorph))
-		infomorph.death(0)
-		infomorph = null
-	QDEL_NULL(radio)
-	return ..()
-
-/obj/item/device/sleevecard/attack_self(mob/user)
-	add_fingerprint(user)
-
-	if(!infomorph)
-		to_chat(user,"<span class='warning'>\The [src] does not have a mind in it!</span>")
-	else
-		to_chat(user,"<span class='notice'>\The [src] displays the name '[infomorph]'.</span>")
-
-//This is a 'hard' proc, it does no permission checking, do that on the computer
-/obj/item/device/sleevecard/proc/sleeveInto(var/datum/transhuman/mind_record/MR, var/db_key)
-	infomorph = new(src,MR.mindname,db_key=db_key)
+/obj/item/device/paicard/attack_ghost(mob/user as mob)
+	return
```

Comment:
Pretty sure that doing this will make it so ghosts can't join unattended pAI cards anymore, which is basically the only way anyone actually plays pAI regularly.

Reply:
Ah heck, that was supposed to be paicard/sleevecard

Reply:
Fixed